src/reactUnusedPropsAndStateRule.ts

```diff
@@ -104,6 +136,109 @@ function walk(ctx: Lint.WalkContext<Options>) {
         return false;
     }
 
+    function inspectPropUsageInObjectBinding(name: ts.ObjectBindingPattern): void {
+        const bindingElements = getObjectBindingData(name);
+        const foundPropNames = Object.keys(bindingElements);
+
+        for (const propName of foundPropNames) {
+            propNames = Utils.remove(propNames, propName);
+        }
+    }
+
+    function lookForReactSpecificArrowFunction(node: ts.TypeReferenceNode): void {
+        const nodeTypeText = node.typeName.getText();
+
+        const isFunctionComponent =
+            nodeTypeText === 'React.SFC' ||
+            nodeTypeText === 'React.FC' ||
+            nodeTypeText === 'React.StatelessComponent' ||
+            nodeTypeText === 'React.FunctionComponent';
```

Comment:
Per the other thread, are you averse to also checking `'SFC'`, `'FC'`, etc.? It's not a blocker, but IMO would be nice.

Reply:
Not a problem at all - I'll add a check for the specific names as well...